packages/affiliates/libs/affiliates.js

```diff
@@ -5,17 +5,21 @@ const highland = require("highland");
 const moment = require("moment");
 const Promise = require("bluebird");
 const assert = require("assert");
+const lodash = require("lodash");
 
 const { getWeb3 } = require("@uma/common");
 const web3 = getWeb3();
 const { toWei, toBN, fromWei } = web3.utils;
 
-const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko, synthPrices }) => {
+const DeployerRewards = ({ queries, empAbi, coingecko, synthPrices, firstEmpDate }) => {
   assert(queries, "requires queries class");
-  assert(empCreatorAbi, "requires creator abi");
   assert(empAbi, "requires empAbi");
   assert(coingecko, "requires coingecko api");
   assert(synthPrices, "requires synthPrices api");
+
+  // use firstEmpDate as a history cutoff when querying for events. We can safely say no emps were deployeed before Jan of 2020.
+  firstEmpDate = firstEmpDate || moment("2020-01-01", "YYYY-MM-DD").valueOf();
```

Comment:
Do you think there's any reason we should look into allowing the user to configure the `firstEmpDate` or is it fine just hardcoding Jan 1?

Reply:
Yea thats a good question. I can imagine if you know the start date of the set of EMPs you are rewarding, you could optimize it a lot by setting the start date manually. Will that ever be done in practice though probably not.